Clean code review

iamr0man opened the pull request.

Commit:
initial commit;

watchlar2000 commented on src/main.js:
```diff
@@ -0,0 +1,14 @@
+
+const app = createApp(App);
+
+app.use(createPinia());
```
Стор ніде не використовується, тому можна видалити строки 2 і 9

watchlar2000 commented on src/api/auth.ts:
```diff
@@ -0,0 +1,18 @@
+}
+
+export function setLocalAccessToken (token: string) {
+  return window.localStorage.setItem("accessToken", token);
```
```
enum Token {
  AccessToken = "accessToken",
}

export function setLocalAccessToken (token: string) {
  return window.localStorage.setItem(Token.AccessToken, token);
}

export function getLocalAccessToken () {
  return window.localStorage.getItem(Token.AccessToken);
}
```

iamr0man replied:
не бачу великої потреби, але винесу

watchlar2000 commented on src/components/icons/IconArrow.vue:
```diff
@@ -0,0 +1,3 @@
+<template>
+  <svg aria-hidden="true" class="w-4 h-4 ml-2 -mr-1" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z" clip-rule="evenodd"></path></svg>
```
Можна відформотувати temlate (як, наприклад, IconMinus)

linkinshurik commented on index.html:
```diff
@@ -0,0 +1,14 @@
+  <body>
+    <div id="app"></div>
+    <script type="module" src="/src/main.js"></script>
+    <script src="https://vjs.zencdn.net/8.0.4/video.min.js"></script>
```
Завантаження в головному індексі бібліотеки - не завжди може підійти, якщо розглянути це зі сторони оптимізації.
Теоретично, в нас може бути випадки, коли користувач не буде використовувати перегляд відео (чи іншу функціональність, лібку якої завантажуємо).
В цьому випадку буде завантаження лишніх данних, іноді суттєвих.
Тож пропоную подумати на майбутнє, як це можна оптимізувати та використовувати лише за необхідності

iamr0man replied:
то для тесту більше було, не використовую цю лібу)

linkinshurik commented on package.json:
```diff
@@ -0,0 +1,50 @@
+		"@vue/test-utils": "^2.3.1",
+		"axios": "^1.3.4",
+		"hls.js": "^1.3.5-0.canary.8977",
+		"jest": "^29.5.0",
```
Варто звернути увагу на пакети, які варто перемістити в devDependencies. Це дрібниця, але дозволить чітко бачити, що в нас потрапить в фінальну збірку

linkinshurik commented on src/components/courses/CourseItem.vue:
```diff
@@ -0,0 +1,173 @@
+          </p>
+          <p class="flex mb-3 font-normal text-gray-700 dark:text-gray-400">
+            {{ course.rating }}
+            <svg width="16" fill="none" stroke="currentColor" stroke-width="1.5" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg" aria-hidden="true">
```
Пропоную винести svg в окремі асети. Це зробить темплейт на багато чистішим

linkinshurik commented on src/components/courses/CourseItem.vue:
```diff
@@ -0,0 +1,173 @@
+  },
+  methods: {
+    playVideo () {
+      const isAvailable = this.course.meta.courseVideoPreview && this.course.meta.courseVideoPreview.duration > 0
```
Схожу логіку зручно було б тримати в computed пропертях. Це б дало можливість спростити функцію і більше зосередитись на логіці

linkinshurik commented on src/components/lesson/Lesson.vue:
```diff
@@ -0,0 +1,210 @@
+
+      return 0;
+    },
+    loadHlsPlayer () {
```
Ця функція виглядає завеликою. В ній відбувається набагато більше процесів, ніж може підсвітити назва методу.
Можливо варто розбити цей метод на декілька, щоб кожен виконував тільки одну завершену дію

Commit:
chore(lint): update package, ts, eslint and prettier configs; add babel;

Commit:
ref: refactor according review;

add types for auth api; add new computed; divide function into small parts; fix lint errors;
